app/grandchallenge/hanging_protocols/forms.py

```diff
@@ -88,6 +88,14 @@ def clean_json(self):
 class ViewContentMixin:
     def clean_view_content(self):
         mapping = self.cleaned_data["view_content"] or {}
+        if not isinstance(mapping, dict):

```

Comment:
Shouldn't we be checking against the schema server side, not just that it is an object?

Reply:
Sorry, completely missed this remark.

Reply:
Do you know why this validation wasn't happening in the first place? It is set on the model field right? So why is it invalid here? Do the model validations just happen later?

Reply:
> I think the model validation happens after the form validates

Can you confirm this?

Reply:
It does, in ModelForms, model.full_clean() is called after the form's clean method.